### PR TITLE
fix(Policies): RHICOMPL-1818- Add notifications for policy actions

### PR DESCRIPTION
**Pull request by bastilian**
This adds notifications for creating and updating policies:

Screenshot 2021-07-01 at 14 03 33

Screenshot 2021-07-01 at 14 02 57

Screenshot 2021-07-01 at 14 01 16

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: src/SmartComponents/CreatePolicy/FinishedCreatePolicy.js [hunk 1]
```diff
@@ -68,10 +69,22 @@ export const FinishedCreatePolicy = ({
         updatePolicy(null, newPolicy, onProgress).then(() => {
             setPercent(100);
             setMessage();
+            dispatchNotification({
```
Comment (bastilian) [inline src/SmartComponents/CreatePolicy/FinishedCreatePolicy.js:72]: I tried extracting this similarly to what is in EditPolicy, to also be able and test it nicely, but it broke a few other things that would have needed more work to get right. 

Generally we should look into changing and moving the updatePolicy call here to a more appropriate place and not run as an effect of a component.

**Review by akofink — COMMENTED** [review]
Editing a policy actually sends a CreateProfile mutation now :/ It looks like there might be multiple issues:

image

**Comment by bastilian** [comment]
@akofink Sorry, I overlooked this. It snuck in when I was dealing with issues in the tests and only appeared with some policies and not all. It should be fixed, the issue was `useCallback`, which didn't have the correct values to pass on.

**Comment by akofink** [comment]
I'm sometimes getting LOTS of notifications when performing a single update action:

image

I'm also seeing a toast notification every time I re-open the modal. In the following screenshot, I deleted a policy from the policies page, saw the toast notification, waited for it to disappear, then used the kebab dropdown to edit another policy, and the old delete notification popped up again. Closing and re-opening the modal makes the notification show every time it's reopened:

image

Same thing if you create a new policy, then edit another policy with the kebab:

image

**Comment by bastilian** [comment]
@akofink Thanks for testing! I could only replicate the issue with extra notifications a handful of times. However, I think what may have happened is that notifications accumulated in the store somehow and got fired with any further notification. It now has `autoDismiss` set, this should reduce the chances of that happening hopefully.

**Review by akofink — COMMENTED** [review]
It works fairly well for the most part, but I'm still seeing notifications appear again after I open the edit policy modal after editing or creating a policy. The weirdness seems to get worse with error notifications. I can show you live tomorrow and see if you are able to reproduce it as well.

https://user-images.githubusercontent.com/761923/125700741-b1857f8d-0ca3-4a73-aec6-5e226bf5b18e.mp4

**Review by akofink — APPROVED** [review]
After meeting to discuss/review the issue, it turned out to be caused by the redux dev tools! Disabling the extension makes this work as expected. Thanks, @bastilian!

**Pull request merged**